astropy/io/fits/header.py

```diff
@@ -93,7 +93,9 @@ def __init__(self, cards=[], copy=False):
 
             .. versionadded:: 1.3
         """
-        self.clear()
+        self._cards = []
```

Comment:
Hm, why this change? Isn't this just duplicating code?

Reply:
I don't think this is a good change. It's identical to `clear` and the method `clear` seems much more obvious to me (even without looking at the method) than these 3 re-assignments.

I'd prefer the previous call `self.clear()` here.